logstash-core/lib/logstash/config/mixin.rb

```diff
@@ -528,6 +530,15 @@ def validate_value(value, validator)
             end
 
             result = value.first.is_a?(::LogStash::Util::SafeURI) ? value.first : ::LogStash::Util::SafeURI.new(value.first)
+          when :uri_list
+            # expand entries that have space-delimited URIs in strings.
+            # This validator is considered private, and can be accessed
+            # by specifying `:validate => :uri` and `:list => true`
+            result = value.flat_map do |entry|
+              entry.kind_of?(String) ? entry.split(' ') : entry
+            end.map do |expanded_entry|
+              ::LogStash::Util::SafeURI.from(expanded_entry)
```

Comment:
@yaauie it's not clear to me in which situation the `expanded_entry` is already an instance of `LogStash::Util::SafeURI`

Reply:
It is a side-effect of `secure_params!`, which is used to modify the _`original_params`_ to obscure sensitive data prior to validation. Essentially by the time this gets to "normal" validation along-side the rest of the params, it has already been processed and will be a list of `SafeURI`.